Sources/YCoreUI/Extensions/UIKit/UIView+constrainCenter.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  UIView+constrainCenter.swift
+//  YCoreUI
+//
+//  Created by Dharmik Ghelani on 20/10/22.
+//  Copyright © 2022 Y Media Labs. All rights reserved.
+//
+
+import UIKit
+
+extension UIView {
+    /// Center alignment options
+    public struct Center: OptionSet {
+        public let rawValue: UInt
+        /// center X
+        public static let x = Center(rawValue: 1 << 0)
+        /// center Y
+        public static let y = Center(rawValue: 1 << 1)
+        /// all (both center X and center Y)
+        public static let all: Center = [.x, .y]
+        // initializer must be declared public to matches a (public init) requirement in protocol 'OptionSet'
```

Comment:
```suggestion

        /// Creates a new `Center` option set from the given raw value.
        /// - Parameter rawValue: the raw value of the `Center` option set to create.
```

again we need doc comments now that this is public.
Let's also insert a single vertical white space between init and the static values.

Reply:
Addressed